models.py

```diff
@@ -276,16 +277,24 @@ def __init__(
 		self.residual = residual
 		self.dict = dict
 		self.bpe_only = bpe_only
+		self.check_time_dim_padded = check_time_dim_padded
 
 	def forward(
 		self, x: typing.Union[shaping.BCT, shaping.BT], xlen: typing.Optional[shaping.B] = None, y: typing.Optional[shaping.BLY] = None, ylen: typing.Optional[shaping.B] = None
 	) -> shaping.BCt:
 
+		if self.check_time_dim_padded:
```

Comment:
otoh, this should be inside the `if`, right to check the input for the frontend, right?

Reply:
Right

Reply:
if you agree, please move this inside the if. it's still correct outside the if but the intent is less clear

Reply:
done